char[] button
float32 trigger
